src/Microsoft.IdentityModel.Xml/Reference.cs

```diff
@@ -161,13 +161,10 @@ private static byte[] ProcessAndDigest(XmlTokenStream tokenStream, HashAlgorithm
         {
             using (var stream = new MemoryStream())
             {
-                using (var writer = XmlDictionaryWriter.CreateTextWriter(Stream.Null))
-                {
-                    tokenStream.WriteTo(writer);
-                    writer.Flush();
-                }
+                using (var writer = XmlWriter.Create(stream, new XmlWriterSettings { CloseOutput = false }))
+                using (var dictionaryWriter = XmlDictionaryWriter.CreateDictionaryWriter(writer))
+                    tokenStream.WriteTo(dictionaryWriter);
 
-                stream.Flush();
```

Comment:
why are the two Flush() calls removed?

Reply:
I assumed it was ok, since XmlTextWriters auto flush when disposed.

Reply:
@gislikonrad 
I tend to assume that sometimes the 'Flush' is not always consistent with different writers and like to be consistent.
For example, I have found Utf8JsonWriter and MemoryStreams have to be managed in certain order when calling Flush.

If you can add the Flush back on the dictionaryWriter, we should be good to go.

Reply:
I'll re-add the flush.

I am, genuinely confused as to what this method did originally. Writing the token stream to a null stream and flushing the writer, then flushing an empty memory stream and hashing nothing?